x-pack/filebeat/input/httpjson/request.go

```diff
@@ -197,15 +371,17 @@ func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, p
 			continue
 		}
 
-		event, err := makeEvent(maybeMsg.msg)
-		if err != nil {
-			r.log.Errorf("error creating event: %v", maybeMsg)
-			continue
-		}
-
-		if err := publisher.Publish(event, trCtx.cursorMap()); err != nil {
-			r.log.Errorf("error publishing event: %v", err)
-			continue
+		if publish {
+			event, err := makeEvent(maybeMsg.msg)
+			if err != nil {
+				r.log.Errorf("error creating event: %v", maybeMsg)
+				continue
+			}
+
+			if err := publisher.Publish(event, trCtx.cursorMap()); err != nil {
+				r.log.Errorf("error publishing event: %v", err)
+				continue
+			}
```

Comment:
Here I am trying to use existing cursor even if response are from chain call, so let's say I want to store value from chain call and use it in main call next time I can do that using cursor. we actually implemented this in salesforce module.
here's [link](https://github.com/kush-elastic/beats/blob/salesforce_observability_connector/x-pack/filebeat/module/salesforce/login-rest/config/login-rest.yml#L33).

Reply:
I am worried this breaks current behavior. In the docs (https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-httpjson.html#input-state) it is mentioned that 

```
last_event: A map representing the last event sent to the output (result from applying transforms to last_response.body)
```

If we update last_event for every chain, regardless of the event being published, this statement is not true anymore, and since cursor can reference last_event, it opens room for potentially confusing situations. 

I see that for your use case, you are only using one chain step, which would make it to behave the same with my suggestion. Do you think is okay to preserve current behavior (and maybe update cursor docs to make clear that updates on published events, also), and maybe we can see how to enhance this if needed in the future?

Reply:
Sorry, My mistake.
But we need this as we are collecting it from first response and using it again in first call.

Reply:
Can't we update doc with like `last_event of specific call`.